src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,66 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
```

Comment:
The goal is to be able to use any [HandleBase](https://docs.rs/magenta/0.1.0/magenta/trait.HandleBase.html) here, so that we can write a tokio-based library using `PollEvented<EventedHandle<T>>`. We need visibility into the handle, although I suppose that could be done using a `get_handle`/`get_handle_mut` methods. The token can be private, too.